src/lib/es6.d.ts

```diff
@@ -3542,6 +3572,16 @@ declare module Reflect {
     function setPrototypeOf(target: any, proto: any): boolean;
 }
 
+interface IPromise<T> {

```

Comment:
According to the Promise/A+ specification, a "thenable" is any object with a callable "then". A "promise" is a "thenable" that conforms to the Promise/A+ specification. In the async functions prototype, I use the `Thenable` interface for type checking a thing that is *not* a promise.

I think a better interface name is `PromiseLike`, similar to our `ArrayLike`.